Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -908,7 +908,12 @@ def _get_x_errorbar_segments(err_cont):
 
 def _get_y_errorbar_segments(err_cont):
     if err_cont.has_yerr and not err_cont.has_xerr:
-        return err_cont[2][0].get_segments()
+        y_errorbar_segments = err_cont[2][0].get_segments()
+        # if y axis contains all nan then
+        if not [i for i in y_errorbar_segments if i.size > 0]:
```

Comment:
The unhandled exception is still happening if some of the values (but not all of them) are NaNs.

 i.e, change line 6 in the issue script to have `DataY='nan,1,nan'` and the original issue still happens. 

Might be worth adding a test for this case as well.

Reply:
That seems sensible, is it worth checking performance? With a large number of bins that seems like it could get really slow if it's having to go through the list many times